conda-recipe/meta.yaml

```diff
@@ -27,29 +28,41 @@ requirements:
         - dpcpp-cpp-rt >={{ required_compiler_version }},!={{ excluded_compiler_version1 }},!={{ excluded_compiler_version2 }},!={{ excluded_compiler_version3 }}  # [win]
         - dpcpp-cpp-rt >={{ required_compiler_version }},!={{ excluded_compiler_version1 }},!={{ excluded_compiler_version2 }}  # [linux]
         - python
-        - setuptools >=63.*
-        - scikit-build >=0.15*
-        - ninja # [not win]
-        - cmake >=3.26*
-        - numba >=0.59*
-        - dpctl >=0.16*
-        - dpnp >=0.14*
-        - dpcpp-llvm-spirv
-        - wheel
+        - setuptools >=69
+        - scikit-build >=0.17
+        - ninja >=1.11.1 # [not win]
+        - cmake >=3.29
+        - numba >=0.59
+        - llvmlite >=0.42.0
+        - dpctl >=0.16
+        - dpnp >=0.14
+        - numpy >=1.24
+        # TODO: there is no 2024 release for python 3.11
+        # - dpcpp-llvm-spirv >={{ required_compiler_version }}
+        - dpcpp-llvm-spirv >=2023.0
+        - wheel >=0.43
+        - pip >=24.0
+        - python-build >=1.1
+        - versioneer==0.29
+        # versioneer dependency
+        - tomli # [py<311]
     run:
         - {{ pin_compatible('dpcpp-cpp-rt', min_pin='x.x', max_pin='x') }}
+        - {{ pin_compatible('intel-cmplr-lib-rt', min_pin='x.x', max_pin='x') }}
+        - {{ pin_compatible('dpcpp-llvm-spirv', min_pin='x.x', max_pin='x') }}
```

Comment:
what does `x.x` imply?

Reply:
This particular line means to support dpcpp from 2024.1 to 2025. You can read more about it here: https://docs.conda.io/projects/conda-build/en/stable/resources/variants.html#pinning-at-the-recipe-level
This is the same approach dpctl and dpnp uses

Reply:
This is how dependencies look like after those resolutions:
```
dependencies: 
  - __glibc >=2.28,<3.0.a0
  - dpcpp-cpp-rt >=2024.1,<2025.0a0
  - dpcpp-llvm-spirv >=2024.1,<2025.0a0
  - dpctl >=0.17.0dev0,<0.18.0a0
  - dpnp >=0.15.0dev1,<0.16.0a0
  - intel-cmplr-lib-rt >=2024.1,<2025.0a0
  - libgcc-ng >=13
  - libstdcxx-ng >=13
  - llvmlite >=0.42.0,<0.43.0a0
  - numba >=0.59.1,<0.60.0a0
  - numpy >=1.26.4,<2.0a0
  - python >=3.10,<3.11.0a0
  - python_abi 3.10.* *_cp310
```